app/controllers/admin/permalinks_controller.rb

```diff
@@ -55,6 +45,24 @@ def fetch_permalink
   end
 
   def permalink_params
-    params.require(:permalink).permit(:url, :permalink_type, :permalink_type_value)
+    permitted_params =
+      params.require(:permalink).permit(:url, :permalink_type, :permalink_type_value)
+
+    {
+      url: permitted_params[:url],
+      topic_id: extract_param(permitted_params, "topic"),
+      post_id: extract_param(permitted_params, "post"),
+      category_id: extract_param(permitted_params, "category"),
+      tag_id:
+        extract_param(permitted_params, "tag").then do |tag_name|
+          (Tag.where(name: tag_name).pluck(:id).first || -1) if tag_name
```

Comment:
Admins input the tag name, but we need to store the `tag_id`. If the tag can't be found, we need to store an invalid ID (`-1`); otherwise the permalink will show up as "topic permalink" in the UI. That's confusing.

We'd have to store the permalink type in the DB if we need a less hacky solution.